protocol/x/vault/keeper/orders.go

```diff
@@ -21,31 +21,33 @@ import (
 func (k Keeper) RefreshAllVaultOrders(ctx sdk.Context) {
 	// Iterate through all vaults.
 	numActiveVaults := 0
-	totalSharesIterator := k.getTotalSharesIterator(ctx)
-	defer totalSharesIterator.Close()
-	for ; totalSharesIterator.Valid(); totalSharesIterator.Next() {
-		vaultId, err := types.GetVaultIdFromStateKey(totalSharesIterator.Key())
+	vaultParamsIterator := k.getVaultParamsIterator(ctx)
+	defer vaultParamsIterator.Close()
+	for ; vaultParamsIterator.Valid(); vaultParamsIterator.Next() {
+		vaultId, err := types.GetVaultIdFromStateKey(vaultParamsIterator.Key())
 		if err != nil {
 			log.ErrorLogWithError(ctx, "Failed to get vault ID from state key", err)
 			continue
 		}
-		var totalShares types.NumShares
-		k.cdc.MustUnmarshal(totalSharesIterator.Value(), &totalShares)
+		var vaultParams types.VaultParams
+		k.cdc.MustUnmarshal(vaultParamsIterator.Value(), &vaultParams)
 
-		// Skip if TotalShares is non-positive.
-		if totalShares.NumShares.Sign() <= 0 {
+		if vaultParams.Status != types.VaultStatus_VAULT_STATUS_QUOTING {
+			// TODO (TRA-546): cancel any existing orders and don't place new orders.
 			continue
 		}
+		// TODO (TRA-547): implement close-only mode.
+
+		// Use default quoting params if no custom ones.
```

Comment:
nit: Might be cleaner to have a separate function the `vaultParam` that takes in a keeper and returns `QuotingParams`, fetching either the default or the non-nil quoting params.